deps/wrapper.cc

```diff
@@ -131,6 +183,44 @@ bool JuliaGcsClient::Exists(const std::string &ns,
     return exists;
 }
 
+ObjectID _submit_task(std::string project_dir) {
+    auto &worker = CoreWorkerProcess::GetCoreWorker();
+
+    RayFunction func(
+        Language::JULIA,
+        FunctionDescriptorBuilder::BuildJulia("", "demo_task", "")
+    );
+
+    std::vector<std::unique_ptr<TaskArg>> args;
+    std::string str = "hello";
+    auto buffer = std::make_shared<LocalMemoryBuffer>(reinterpret_cast<uint8_t *>(&str[0]), str.size(), true);
+    auto ray_obj = std::make_shared<RayObject>(buffer, nullptr, std::vector<rpc::ObjectReference>());
+    args.emplace_back(new TaskArgByValue(ray_obj));
+
+    TaskOptions options;
+
+    // TaskOptions: https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/core_worker/common.h#L62-L87
+    // RuntimeEnvInfo (`options.serialized_runtime_env_info`): https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/protobuf/runtime_env_common.proto#L39-L46
+    // RuntimeEnvContext (`{"serialized_runtime_env": ...}`): https://github.com/ray-project/ray/blob/ray-2.5.1/python/ray/_private/runtime_env/context.py#L20-L45
+    options.serialized_runtime_env_info = "{\"serialized_runtime_env\": \"{\\\"env_vars\\\": {\\\"JULIA_PROJECT\\\": \\\"" + project_dir + "\\\"}}\"}";
```

Comment:
I think that eventually we'll want to hoist this all into the actual julia runtime/worker code right?

Reply:
Ideally most of this would be in Julia. However, as all the C++ classes in Julia are heap allocated there may be some performance reasons to keep some of this on the C++ side of the fence.